pkg/beaconsni/mongodb.go

```diff
@@ -0,0 +1,193 @@
+package beaconsni
+
+import (
+	"runtime"
+
+	"github.com/activecm/rita/config"
+	"github.com/activecm/rita/database"
+	"github.com/activecm/rita/pkg/data"
+	"github.com/activecm/rita/pkg/host"
+	"github.com/activecm/rita/pkg/sniconn"
+	"github.com/activecm/rita/util"
+	"github.com/globalsign/mgo"
+	"github.com/vbauerster/mpb"
+	"github.com/vbauerster/mpb/decor"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type repo struct {
+	database *database.DB
+	config   *config.Config
+	log      *log.Logger
+}
+
+//NewMongoRepository bundles the given resources for updating MongoDB with SNI connection data
+func NewMongoRepository(db *database.DB, conf *config.Config, logger *log.Logger) Repository {
+	return &repo{
+		database: db,
+		config:   conf,
+		log:      logger,
+	}
+}
+
+// CreateIndexes creates indexes for the beaconSNI collection
+func (r *repo) CreateIndexes() error {
+
+	session := r.database.Session.Copy()
+	defer session.Close()
+
+	// set collection name
+	collectionName := r.config.T.BeaconSNI.BeaconSNITable
+
+	// check if collection already exists
+	names, _ := session.DB(r.database.GetSelectedDB()).CollectionNames()
+
+	// if collection exists, we don't need to do anything else
+	for _, name := range names {
+		if name == collectionName {
+			return nil
+		}
+	}
+
+	// set desired indexes
+	indexes := []mgo.Index{
+		{Key: []string{"-score"}},
+		{Key: []string{"src", "fqdn", "src_network_uuid"}, Unique: true},
+		{Key: []string{"src", "src_network_uuid"}},
+		{Key: []string{"fqdn"}},
+		{Key: []string{"responding_ips.ip", "responding_ips.network_uuid"}},
```

Comment:
We do not create indexes for `resolved_ips` in `beaconFQDN`. Do you remember any contradiction for adding them in `beaconFQDN` or was this just an oversight? I have a faint memory of the index causing a slowdown instead of a speed increase, but I seriously could have been dreaming.

Reply:
I'll go take a brief skim through our past tickets, but this doesn't ring a bell for me. I don't think I was heavily involved with the development of `beaconFQDN`, so I might not be the best person to ask though.

Reply:
I couldn't find anything related to it while sifting through old issues & PRs, so I'll just blame it on the mandela effect.